lib/core/src/Cardano/Wallet.hs

```diff
@@ -521,7 +528,7 @@ newWalletLayer tracer block0 feePolicy db nw tl = do
         -> BlockHeader
         -> IO ()
     restoreSleep t wid slot = do
-        let tenSeconds = 10000000 in threadDelay tenSeconds
+        let halfSlotLengthDelay = 500000 * (fromIntegral slotLength) in threadDelay halfSlotLengthDelay
```

Comment:
For readability, could we separate
- Converting seconds to microseconds
- Dividing by two
?

I'm not sure what our preferred way of doing the conversion is though. For _literals_ I like the `1 * second` style.

Reply:
I wish we could simply have something like

```hs
{-# LANGUAGE NamedTimeConstants #-}

{- ... -}
threadDelay 30ms
{-  ... -}
```

That would be... fantastic. One thing that's also possible but haven't quite agreed with myself about whether it's clever or an abomination:

```hs
data TimeUnit = Seconds | Milliseconds | Microseconds

toScale :: TimeUnit -> Int
toScale = \case 
    Seconds -> 1000000
    Milliseconds -> 1000
    Microseconds -> 1

instance Num (TimeUnit -> Int) where
    fromInteger n = \u -> fromEnum n * toScale u

>>> 20 Milliseconds :: Int
20000

>>> threadDelay (100 Milliseconds) *> putStrLn "patate"
{- ... 100ms ... -}
patate

>>> threadDelay (2 Seconds) *> putStrLn "patate"
{- ... 2s ... -}
patate
```

Reply:
This one I will leave as it is - I bet we are going to deliver here something big soon :-)